openmmml/models/nequippotential.py

```diff
@@ -0,0 +1,198 @@
+
+                self.model, metadata = nequip.scripts.deploy.load_deployed_model(model_path, freeze=False)
+
+                self.default_dtype= {"float32": torch.float32, "float64": torch.float64}[metadata["model_dtype"]]
+                torch.set_default_dtype(self.default_dtype)
+
+                self.register_buffer('r_max', torch.tensor(float(metadata["r_max"])))
+                
+                if atom_types is not None: # use user set explicit atom types
+                    nequip_types = atom_types
+                
+                else: # use openmm atomic symbols
+
+                    type_names = str(metadata["type_names"]).split(" ")
+
+                    type_name_to_type_index={ type_name : i for i,type_name in enumerate(type_names)}
+
+                    nequip_types = [ type_name_to_type_index[atom.element.symbol] for atom in includedAtoms]
+                
+                atomic_numbers = [atom.element.atomic_number for atom in includedAtoms]
+
+                self.atomic_numbers = torch.nn.Parameter(torch.tensor(atomic_numbers, dtype=torch.long), requires_grad=False)
+                #self.N = len(includedAtoms)
+                self.atom_types = torch.nn.Parameter(torch.tensor(nequip_types, dtype=torch.long), requires_grad=False)
+
+                if periodic:
+                    self.pbc = torch.nn.Parameter(torch.tensor([True, True, True]), requires_grad=False)
+                else:
+                    self.pbc = torch.nn.Parameter(torch.tensor([False, False, False]), requires_grad=False)
+
+                # indices for ML atoms in a mixed system
+                if indices is None: # default all atoms are ML
+                    self.indices = None
+                else:
+                    self.indices = torch.tensor(indices, dtype=torch.int64)
+
+
+            def forward(self, positions, boxvectors: Optional[torch.Tensor] = None):
+                
+                # setup positions
+                positions = positions.to(dtype=self.default_dtype)
+                if self.indices is not None:
+                    positions = positions[self.indices]
+                positions = positions*self.nm_to_distance
+
+
+                # prepare input dict 
+                input_dict={}
+
+                if boxvectors is not None:
+                    input_dict["cell"]=boxvectors.to(dtype=self.default_dtype) * self.nm_to_distance
+                    pbc = True
+                else:
+                    input_dict["cell"]=torch.eye(3, device=positions.device)
+                    pbc = False
+
+                input_dict["pbc"]=self.pbc
+                input_dict["atomic_numbers"] = self.atomic_numbers
+                input_dict["atom_types"] = self.atom_types
+                input_dict["pos"] = positions
+
+                # compute edges
+                mapping, shifts_idx = simple_nl(positions, input_dict["cell"], pbc, self.r_max)
```

Comment:
Is there a reason not to use the NNPOps neighbor list?  In https://github.com/mir-group/nequip/issues/288#issuecomment-1475964360 you found it was much faster.  NNPOps is already a dependency of this module.